Example/PrebidInternalTestApp/src/main/java/org/prebid/mobile/renderingtestapp/utils/ThirdPartyRenderer.java

```diff
@@ -0,0 +1,159 @@
+
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+import static org.prebid.mobile.api.exceptions.AdException.THIRD_PARTY;
+
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+
+import org.prebid.mobile.api.exceptions.AdException;
+import org.prebid.mobile.api.rendering.customrenderer.CustomBannerRenderer;
+import org.prebid.mobile.api.rendering.customrenderer.CustomInterstitialRenderer;
+import org.prebid.mobile.api.rendering.customrenderer.InterstitialControllerInterface;
+import org.prebid.mobile.configuration.AdUnitConfiguration;
+import org.prebid.mobile.rendering.bidding.data.bid.BidResponse;
+import org.prebid.mobile.rendering.bidding.interfaces.InterstitialControllerListener;
+import org.prebid.mobile.rendering.bidding.listeners.DisplayViewListener;
+
+import tv.teads.sdk.AdOpportunityTrackerView;
+import tv.teads.sdk.AdPlacementSettings;
+import tv.teads.sdk.AdRatio;
+import tv.teads.sdk.AdRequestSettings;
+import tv.teads.sdk.InReadAd;
+import tv.teads.sdk.InReadAdModelListener;
+import tv.teads.sdk.InReadAdPlacement;
+import tv.teads.sdk.TeadsSDK;
+import tv.teads.sdk.renderer.InReadAdView;
+
+// TODO temp for tests purposes, should not be merged
+public class ThirdPartyRenderer implements CustomBannerRenderer, CustomInterstitialRenderer {
+
+    private FrameLayout frameLayout;
+
+    @Override
+    public View getBannerAdView( @NonNull Context context,
+                                 DisplayViewListener listener,
+                                 @NonNull AdUnitConfiguration adUnitConfiguration,
+                                 @NonNull BidResponse response) {
+        frameLayout = new FrameLayout(context);
+
+        InReadAdView inReadAdView = new InReadAdView(context);
+        inReadAdView.setLayoutParams(new ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT));
+        getAdd(inReadAdView, listener);
+
+        return inReadAdView;
+    }
+
+    @Override
+    public InterstitialControllerInterface getInterstitialController(Context context, InterstitialControllerListener listener) {
+        return new InterstitialControllerInterface() {
+            @Override
+            public void loadAd(AdUnitConfiguration adUnitConfiguration, BidResponse bidResponse) {
+                Toast.makeText(context, "Load Interstitial Ad", Toast.LENGTH_LONG).show();
+                listener.onInterstitialReadyForDisplay();
+            }
+
+            @Override
+            public void show() {
+                Toast.makeText(context, "Show Interstitial Ad", Toast.LENGTH_LONG).show();
+                listener.onInterstitialDisplayed();
+            }
+
+            @Override
+            public void destroy() {
+
+            }
+        };
+    }
+
+    private void getAdd(InReadAdView inReadAdView, DisplayViewListener displayViewListener) {
```

Comment:
Hey @YuriyVelichkoPI we share of same comprehension. This is just for tests purposes since we don't have yet the necessary data in the bid response to just delegate the rendering, then for now, we make the ad request. We understand that all data we need to render the ad will come with bid response without requiring new requests. 

And sure you possibly saw the "no ad" error cause of this temporary approach. But please, don't took it in consideration. This is temporary and will be removed in next commits. It will be replaced by a calling function which just return an ad view using the existing data in the bid response.

Is it more clear now?

Reply:
yep, just want to be sure that everyone on the same page! Thanks!